### PR TITLE
Add `QuickwitService`s to `QuickwitConfig`

### PR DESCRIPTION
**Pull request by boraarslan**
This PR just adds the service field to the config. It currently does not have any functionality.

### CODE REVIEW

File: quickwit-config/src/config.rs [hunk 1]
```diff
@@ -77,6 +79,13 @@ fn default_rest_listen_port() -> u16 {
     7280
 }
 
+fn default_quickwit_services() -> Vec<String> {
+    vec!["metastore", "searcher", "indexer"]
```
Comment (guilload) [inline quickwit-config/src/config.rs:83]: Let's add `janitor`.

File: quickwit-config/src/config.rs [hunk 2]
```diff
@@ -77,6 +79,13 @@ fn default_rest_listen_port() -> u16 {
...
+        .iter()
+        .map(|s| s.to_string())
```
Comment (guilload) [inline quickwit-config/src/config.rs:85]:
```suggestion
        .map(|svc| svc.to_string())
```

File: quickwit-config/src/config.rs (hunk 1 above)
Comment (guilload) [inline quickwit-config/src/config.rs:83]: You can directly use an array.

File: quickwit-config/src/config.rs [hunk 3]
```diff
@@ -228,6 +239,13 @@ impl QuickwitConfigBuilder {
         serde_yaml::from_slice(bytes).context("Failed to parse YAML config file.")
     }
 
+    fn services_from_vec(&self) -> anyhow::Result<HashSet<QuickwitService>> {
```
Comment (guilload) [inline quickwit-config/src/config.rs:242]:
```suggestion
    fn parse_services(&self) -> anyhow::Result<HashSet<QuickwitService>> {
```

File: quickwit-common/src/service.rs [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        match self {
+            QuickwitService::Indexer => "indexer",
+            QuickwitService::Searcher => "searcher",
+            QuickwitService::Janitor => "janitor",
+            QuickwitService::Metastore => "metastore",
+        }
+    }
+}
+
+impl TryFrom<&str> for QuickwitService {
```
Comment (guilload) [inline quickwit-common/src/service.rs:42]: That's basically `FromStr`.

File: quickwit-common/src/service.rs [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    type Error = anyhow::Error;
+
+    fn try_from(service_str: &str) -> Result<Self, Self::Error> {
+        match service_str {
+            "indexer" => Ok(QuickwitService::Indexer),
+            "searcher" => Ok(QuickwitService::Searcher),
+            "janitor" => Ok(QuickwitService::Janitor),
+            "metastore" => Ok(QuickwitService::Metastore),
+            _ => {
+                bail!("Service `{service_str}` unknown");
```
Comment (guilload) [inline quickwit-common/src/service.rs:52]:
```rust
bail!("Failed to parse service `{service_str}`. Supported services are: {}", QuickwitService::supported_services().join(", ")
```

File: quickwit-cli/src/main.rs [hunk 6]
```diff
@@ -25,9 +25,9 @@ use opentelemetry::global;
 use opentelemetry::sdk::propagation::TraceContextPropagator;
 use quickwit_cli::cli::{build_cli, CliCommand};
 use quickwit_cli::QW_JAEGER_ENABLED_ENV_KEY;
-use quickwit_cluster::QuickwitService;
 use quickwit_common::metrics::new_gauge;
 use quickwit_common::runtimes::RuntimesConfiguration;
+use quickwit_common::service::QuickwitService;
```
Comment (guilload) [inline quickwit-cli/src/main.rs:30]: I want to keep `quickwit-common` free of Quickwit-specific business logic. So let's move this code to `quickwit-config` instead.

File: quickwit-config/src/config.rs [hunk 7]
```diff
@@ -494,13 +514,22 @@ impl QuickwitConfig {
         let metastore_uri = default_metastore_uri(&data_dir_path);
         let default_index_root_uri = default_index_root_uri(&data_dir_path);
 
+        let services = default_quickwit_services()
```
Comment (guilload) [inline quickwit-config/src/config.rs:517]:
```rust
let services = default_quickwit_services()
            .iter()
            .map(|svc| svc.parse())
            .collect::<anyhow::Result<_>>()
            .expect("Default services should be parsed correctly.");
```

File: quickwit-config/src/config.rs [hunk 8]
```diff
@@ -171,6 +180,8 @@ struct QuickwitConfigBuilder {
     advertise_address: Option<String>,
     #[serde(default = "default_rest_listen_port")]
     rest_listen_port: u16,
+    #[serde(default = "default_quickwit_services")]
+    services: Vec<String>,
```
Comment (guilload) [inline quickwit-config/src/config.rs:184]: @fmassot, I'm also considering `enabled_services`, WDYT?

**Review by guilload — APPROVED** [review]
Approved, but please, address my last comments.

File: quickwit-config/src/config.rs (hunk 2 above)
Comment (fmassot) [reply quickwit-config/src/config.rs]: `service` is fine too.

File: quickwit-config/src/config.rs (hunk 8 above)
Comment (guilload) [reply quickwit-config/src/config.rs:185]: Chatted with @fmassot, let's go with `enabled_services`.

**Pull request merged**